### PR TITLE
Saving intermediate camera images 

### PR DESCRIPTION
**Pull request by RaresAmbrus**
Added functionality to save camera images (rgb & depth) acquired at intermediate positions during a sweep to disk, as well as logging the TF position of the Depth camera and the rgb camera at each position along with the camera parameters in the sweep xml file. 

This information is useful for registration of individual point clouds acquired during a sweep, estimating motion in the scene at a particular sweep position, estimating a per pixel noise model as well as  rgb <-> depth camera calibration (intrinsic and extrinsic). 

Loading of the new data from the xml as well as the saved images is also supported by the metaroom_xml_parser (standalone parser for using the sweep data).

### CODE REVIEW

File: cloud_merge/launch/cloud_merge.launch [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <launch>
 
-  <arg name="save_intermediate"           default="true" />
+  <arg name="save_intermediate_clouds"    default="true" />
```
Comment (marc-hanheide) [inline cloud_merge/launch/cloud_merge.launch:3]: Did you check that any of our other launch files reflect this change? I want to be much more rigorous about changes that can make our launch files fail. Also, I don't think it is a good idea for a production environment to set this to true by default.

Comment (marc-hanheide) [inline cloud_merge/launch/cloud_merge.launch:3]: This applies to all the changes in this file

**Comment by marc-hanheide** [comment]
did anyone test this on the robot?

**Comment by RaresAmbrus** [comment]
There is only one additional launch file:

```
https://github.com/RaresAmbrus/strands_3d_mapping/blob/log_images/cloud_merge/launch/mapping.launch
```

I have made sure the changes are reflected there as well.

Regarding 

```
-  <arg name="save_intermediate"           default="true" />
+  <arg name="save_intermediate_clouds"    default="true" />
```

I have only renamed this parameter, its value was always true. For now we need to log this data, it can be used to generate everything else and we need it, not just for the static vs dynamic analysis but also for @Pandoro's semantic labeling component.

Note that this pull request relates to also saving intermediate images, and that parameter is set to false: https://github.com/RaresAmbrus/strands_3d_mapping/blob/log_images/cloud_merge/launch/mapping.launch#L4

I have tested this on the robot and recorded a couple of sweeps. If there are issues I will fix them later on, but for now I think you should merge this as it makes work difficult on my branches since I'm still working on these files for other features.

**Comment by RaresAmbrus** [comment]
@marc-hanheide can you please merge these pull requests? Or perhaps it would be easier if you gave me write access to this repository for now.

**Comment by hawesie** [comment]
@RaresAmbrus there's no way we can judge whether a PR of this size is ok and safe to merge for the rest of the project. To get around this we will give you write access to this repo. However we would like to see some unit tests in your code that can be automatically run before we start re-releasing stuff.

**Comment by RaresAmbrus** [comment]
@hawesie: as I have tested my PRs on the robot, I think it's safe to merge them. If you can point me to standard instructions for setting up unit tests and usage examples in STRANDS I would be grateful and try to add them as well, but to be honest it's the first time I hear about unit tests in relation to a pull request or a release in Strands. Since this code is running on our robot in combination with the latest marathon setup I assumed it was safe to merge it back to the trunk.

**Comment by hawesie** [comment]
This is the first time you’ve heard about it because @marc-hanheide and myself only really started discussing this yesterday (in response to some of the failures across the project in the marathon). 

Check out http://wiki.ros.org/rostest which will create tests that will be automatically run by Jenkins on a PR.

There are tests in packages in the mongodb_store and strands_executive repos if you want to see what other people have done.=

**Pull request merged**